infrastructure/resource_group_level/modules/webapps/gef-ui.bicep

```diff
@@ -0,0 +1,139 @@
+param additionalSecureSettings object = {
+  DOCKER_REGISTRY_SERVER_PASSWORD: 'test-value'
+  DTFS_CENTRAL_API_KEY: 'test-value'
+  EXTERNAL_API_KEY: 'test-value'
+  PORTAL_API_KEY: 'test-value'
+  TFM_API_KEY: 'test-value'
+}
+
+// These values are taken from GitHub secrets injected in the GHA Action
+@secure()
+param secureConnectionStrings object = {
+  // TODO:FN-820 Remove COMPANIES_HOUSE_API_KEY as it is not referenced directly in gef-ui
+  COMPANIES_HOUSE_API_KEY : 'test-value'
+  SESSION_SECRET: 'test-value'
+}
+
+// These values are taken from an export of Connection strings on Dev (& validating with staging).
+@secure()
+param additionalSecureConnectionStrings object = {
+}
+
+var dockerImageName = '${containerRegistryName}.azurecr.io/${resourceNameFragment}:${environment}'
+var dockerRegistryServerUsername = 'tfs${environment}'
+
+resource redis 'Microsoft.Cache/redis@2022-06-01' existing = {
+  name: redisName
+}
+
+var portalApiUrl = 'https://${portalApiHostname}'
+// Note that for externalApiUrl in the CLI script, http was used, but the value in the exported config was https.
+var externalApiUrl = 'https://${externalApiHostname}'
+
+
+// https://learn.microsoft.com/en-us/azure/virtual-network/what-is-ip-address-168-63-129-16
+var azureDnsServerIp = '168.63.129.16'
+
+// These values are hardcoded in the CLI scripts, derived in the script or set from normal env variables
+var settings = {
+  // from env.
+  // TODO:FN-820 Remove COMPANIES_HOUSE_API_URL as it is not referenced directly in gef-ui
+  COMPANIES_HOUSE_API_URL: 'test-value'
+
+  // derived
+  PORTAL_API_URL: portalApiUrl
+  REDIS_HOSTNAME: redis.properties.hostName
+  REDIS_PORT: redis.properties.sslPort
+  REDIS_KEY: redis.listKeys().primaryKey
+  EXTERNAL_API_URL: externalApiUrl
+  HTTPS: 1
+
+  // hard coded
+  WEBSITE_DNS_SERVER: azureDnsServerIp
+  WEBSITE_VNET_ROUTE_ALL: '1'
+  PORT: '5000'
+  WEBSITES_PORT: '5000'
+}
+
+// These values are taken from an export of Configuration on Dev (& validating with staging).
+var tfmApiUrl = 'https://${tfmApiHostname}'
+
+var additionalSettings = {
+  DEAL_API_URL: portalApiUrl // TODO:FN-805 remove DEAL_API_URL
+
+  DOCKER_ENABLE_CI: 'true'
+  DOCKER_REGISTRY_SERVER_URL: '${containerRegistryName}.azurecr.io'
+  DOCKER_REGISTRY_SERVER_USERNAME: dockerRegistryServerUsername
+  LOG4J_FORMAT_MSG_NO_LOOKUPS: 'true'
+  TFM_API_URL: tfmApiUrl
+  TZ: 'Europe/London'
+  WEBSITE_HTTPLOGGING_RETENTION_DAYS: '3'
+  WEBSITES_ENABLE_APP_SERVICE_STORAGE: 'false'
+}
+
+var nodeEnv = nodeDeveloperMode ? { NODE_ENV: 'development' } : {}
+
+var appSettings = union(settings, secureSettings, additionalSettings, additionalSecureSettings, nodeEnv)
+
+var connectionStringsList = [for item in items(union(secureConnectionStrings, additionalSecureConnectionStrings)): {
+  name: item.key
+  value: item.value
+} ]
+
+var connectionStringsProperties = toObject(connectionStringsList, item => item.name, item => {
+  type: 'Custom'
+  value: item.value
+} )
+
```

Comment:
Note that there is still a lot of overlap between the different webapp files.
I think it's probably best to see if I can commonise further when I've sorted out how the parameters (including secure ones) get injected.
 (e.g. I could have one big `webapps` file which works out all the derived values and created instances using the current `webapp` submodule.)